File: data-warehouse-web/src/classes/Pokemon/PokemonEvent.ts
import { AuditInfo } from "../Common/AuditInfo";

export class PokemonEvent {

    public eventID: number;
    public isEventActive: boolean;
    public eventName: string;
    public eventType: string;
    public startDate: Date;
    public endDate: Date;
    public serialCode: string;
    public teraType: string;
    public description: string;
    public isNew: boolean;
    public auditInfo: AuditInfo;
    
    static emptyEvent: PokemonEvent = {
        eventID: -1,
        isEventActive : true,
        eventName: '',
        eventType: '',
        startDate: new Date(),
        endDate: new Date(),
        serialCode: '',
        teraType: '',
        description: '',
        isNew: false,
        auditInfo: new AuditInfo(new Date(), '', new Date(), '', false)
    };

    constructor(inEventID: number,
                inIsEventActive: boolean,
                inEventName: string,
                inEventType: string,
                inStartDate: Date,
                inEndDate: Date,
                inSerialCode: string,
                inTeraType: string,
                inDescription: string,
                inIsNew: boolean,
                inCreatedDate: Date,
                inCreatedBy: string,
                inUpdatedDate: Date,
                inUpdatedBy: string,
                inIsDeleted: boolean) {
        this.eventID = inEventID;
        this.isEventActive = inIsEventActive;
        this.eventName = inEventName;
        this.eventType = inEventType;
        this.startDate = inStartDate;
        this.endDate = inEndDate;
        this.serialCode = inSerialCode;
        this.teraType = inTeraType;
        this.description = inDescription;
        this.isNew = inIsNew;
        this.auditInfo = new AuditInfo(inCreatedDate, inCreatedBy, inUpdatedDate, inUpdatedBy, inIsDeleted);
    }
}

File: DataWarehouse/DataWarehouse/Library/Classes/Pokemon/PokemonEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using DataWarehouse.Library.Classes.Common;

namespace DataWarehouse.Library.Classes.Pokemon
{
    public class PokemonEvent
    {

        public int EventID { get; set; }
        public bool IsEventActive { get; set; }
        public string EventName { get; set; }
        public string EventType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string SerialCode { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public AuditInfo AuditInfo { get; set; }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="inEventID"></param>
        /// <param name="inIsEventActive"></param>
        /// <param name="inEventName"></param>
        /// <param name="inEventType"></param>
        /// <param name="inStartDate"></param>
        /// <param name="inEndDate"></param>
        /// <param name="inSerialCode"></param>
        /// <param name="inDescription"></param>
        /// <param name="inAuditInfo"></param>
        public PokemonEvent(int inEventID, bool inIsEventActive, string inEventName, string inEventType, DateTime inStartDate, DateTime inEndDate,
                            string inSerialCode, string inDescription, AuditInfo inAuditInfo)
        {
            EventID = inEventID;
            IsEventActive = inIsEventActive;
            EventName = inEventName;
            EventType = inEventType;
            StartDate = inStartDate;
            EndDate = inEndDate;
            SerialCode = inSerialCode;
            Description = inDescription;
            AuditInfo = new AuditInfo(inAuditInfo);
        }

        /// <summary>
        /// Default Constructor
        /// </summary>
        public PokemonEvent()
            : this(0, true, string.Empty, string.Empty, DateTime.UtcNow, DateTime.UtcNow, string.Empty, string.Empty, new AuditInfo())
        {
            // intentionally left blank
        }

        /// <summary>
        /// Copy Constructor
        /// </summary>
        /// <param name="inCopy"></param>
        public PokemonEvent(PokemonEvent inCopy)
            : this(inCopy.EventID, inCopy.IsEventActive, inCopy.EventName, inCopy.EventType,
                    inCopy.StartDate, inCopy.EndDate, inCopy.SerialCode, inCopy.Description, inCopy.AuditInfo)
        {
            // intentionally left blank
        }
    }
}